### PR TITLE
feat(*): add support for entry aliasing

### LINKED ISSUE #60: Support cross-references/aliases
I was thinking it would be good to add support for cross-references for entries. For example, if you define "running" as entry, but really just want it to link back to "run". This would allow conjugated verbs to be looked up and traced back to their base forms.

I'm thinking the syntax could look like the following:

```
<entry term="running" see="run" />
```

and then the CLI could support a `--follow` or `-f `flag that would just display the original term if you look up a term that references it. 

The caveat here is answering what will happen if an `entry` has both a `see` attribute and listed etymologies. I suppose a client could always just display "See also: <word>" at the top of the definition list.

### PR DESCRIPTION
**Pull request by Nickersoft**
This PR introduces a new attribute to `<entry>` fields: `see`. 

"See" refers to "see also" or "see more", and is intended to link an entry to an associated entry. An example might be if you looked up the conjugated form of a word in a dictionary, and the dictionary advised you to look at the base form instead. Right now, there is only support for one alias per entry. While entries with aliases can technically have etymologies, it's usually unlikely they would.

Also in this PR (things I needed to fix to get the project to build): I upgraded Poetry, along with FlatBuffers to use built-in lookupByKey().

Closes #60.

### CODE REVIEW

**Comment by Nickersoft** [comment]
@maxdemaio curious what your thoughts are regarding this, especially the naming. Something else that feels kind of difficult to pull off, but something I was wondering was whether the JSON returned from `lookup()` should embed the actual Entry of the `see` value or not. Right now it just shows the text, _but_ if you pass a `--follow` flag to the CLI, it will show the entry for `see` instead of the original entry. Not sure if this is too confusing or not. Felt like the simplest way tbh haha.

**Comment by maxdemaio** [comment]
This `see` attribute is amazing. This way, any conjugated verb links back its infinitive form. This makes a lot of sense to me for French and English. I think this this would apply well to other languages too.

I read a bit on how Wiktionary does it. Not that we have to copy what they do, but they seem like a good source to compare against. They do a 1-to-Many relationship:

- https://en.wiktionary.org/wiki/Wiktionary:Entry_layout
- https://en.wiktionary.org/wiki/Template:also

For the CLI, that flag seems awesome too. The dictionary user could have a way to pull that information back when searching. Also nice addition to update those libraries to use built-in functions!

File: jvm/lib/src/test/kotlin/org/odict/DictionaryTest.kt [hunk 1]
```diff
@@ -37,7 +37,7 @@ class DictionaryTest {
                 "test.odict")
         val dict = Dictionary("test.odict")
         val entries = dict.lookup("hello")
-
+println(entries)
```
Comment (maxdemaio) [inline jvm/lib/src/test/kotlin/org/odict/DictionaryTest.kt:40]: Nit: rm console log

**Review by maxdemaio — CHANGES_REQUESTED** [review]
LGTM! Very nice addition to the test cases. Just one quick comment about the `println` and should be good 💪🏻

File: jvm/lib/src/test/kotlin/org/odict/DictionaryTest.kt (hunk 1 above)
Comment (Nickersoft) [reply jvm/lib/src/test/kotlin/org/odict/DictionaryTest.kt:40]: Gah I thought I caught all of them 😆  Thanks!

**Comment by Nickersoft** [comment]
@maxdemaio Fascinating! Yeah this is technically many-to-one, albeit in a different sense, as in _many_ entries can be traced back to _one_ root word. Wiktionary allow you to specify several alternate entries for a single entry. 

I do like `also` though, maybe over `see` – `see` feels a little ambiguous, whereas `also` implies there is _also_ a similar entry. Only issue with supporting multiple `also` tags for a single entry is the behavior of `--follow` then becomes more ambiguous in terms of which definition it would take you. Perhaps `<also>` could be a tag that co-exists with a top-level singular `see` or `alias` attribute. In fact, I think I'll change `see` to `alias` because it's more straightforward, unless you feel strongly.

**Comment by Nickersoft** [comment]
Also, unrelated, but it's annoying how Wiktionary templates are structured, but not structured enough for us to do a direct port of it to ODict lol

**Review by maxdemaio — APPROVED** [review]

**Comment by maxdemaio** [comment]
> Also, unrelated, but it's annoying how Wiktionary templates are structured, but not structured enough for us to do a direct port of it to ODict lol

Lol 100% agreed. I'm going to approve these changes!

Looks like multiple `see` attributes won't work like you mentioned: https://stackoverflow.com/questions/12664154/how-to-add-the-same-attribute-multiple-times-to-an-element-tag-in-xml. Honestly `,` or`|` both sound like valid options 😁

**Pull request merged**